sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java

```diff
@@ -81,7 +81,13 @@ void populatePartitionedQueryMetrics() {
                                 fetchExecutionRangeAccumulator.getExecutionRanges(),
                                 Arrays.asList(schedulingTimeSpanMap)
                         ), pageResult.getActivityId());
-                BridgeInternal.putQueryMetricsIntoMap(pageResult, feedRange.getRange().toString(), qm);
+                String pkrId = pageResult.getResponseHeaders().get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID);
```

Comment:
why can't we rely on the targetRange please see here:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java#L93

is that going be different than this one from the header response?

Reply:
We need both feedrange and rangeId in query metric key , it will be easier to diagnose issue for On call, discussed with @mbhaskar please check this comment #22202 (comment)